File: Python/Longest Palindromic Substring.py
class Solution:
    # @return a string
    def longestPalindrome(self, s):
        n = len(s)
        if n in [0, 1]:
            return s

        maxLen=1
        start=0
        for i in range(n):
            if i-maxLen >=1 and s[i-maxLen-1:i+1]==s[i-maxLen-1:i+1][::-1]:
                start=i-maxLen-1
                maxLen+=2
                continue

            if i-maxLen >=0 and s[i-maxLen:i+1]==s[i-maxLen:i+1][::-1]:
                start=i-maxLen
                maxLen+=1
        return s[start:start+maxLen]

File: Java/src/Longest Palindromic Substring.java
public class Solution {
    public String longestPalindrome(String s) {
        if(s==null) return s;
        int maxLen = 1;
        int start = 0;
        for(int i=0; i< s.length(); i++){
            if (i-maxLen >=1 && isPalindrome(s.substring(i-maxLen-1,i+1))){
                start=i-maxLen-1;
                maxLen+=2;
                continue;
         }   

            if (i-maxLen >=0 && isPalindrome(s.substring(i-maxLen,i+1))){
                start=i-maxLen;
                maxLen+=1;
            }
        }
        
        return s.substring(start, start+maxLen);
        
    }
    public Boolean isPalindrome(String s){
        String reversed = new StringBuilder(s).reverse().toString();
        return s.equals(reversed);
    }
}